pysqldb3/geopackage.py

```diff
@@ -0,0 +1,500 @@
+                        tbl_name = self.table,
+                        srid=self.srid,
+                        gdal_data=self.gdal_data_loc
+                    )
+                else:
+                    self.cmd = WRITE_GPKG_CMD_MS.format(export_path=self.path,
+                                                       gpkg_name=self.name_extension(self.gpkg_name),
+                                                       host=self.dbo.server,
+                                                       username=self.dbo.user,
+                                                       db=self.dbo.database,
+                                                       password=self.dbo.password,
+                                                       ms_sql_select=qry,
+                                                       tbl_name = self.table,
+                                                       srid=self.srid,
+                                                       gdal_data=self.gdal_data_loc)
+
+        if print_cmd:
+            print(print_cmd_string([self.dbo.password], self.cmd))
+
+        try:
+            ogr_response = subprocess.check_output(shlex.split(self.cmd.replace('\n', ' ')), stderr=subprocess.STDOUT)
+            print(ogr_response)
+        except subprocess.CalledProcessError as e:
+            print("Ogr2ogr Output:\n", e.output)
+            print('Ogr2ogr command failed. The Geopackage/feature class was not written.')
+            raise subprocess.CalledProcessError(cmd=print_cmd_string([self.dbo.password], self.cmd), returncode=1)
+
+        if self.table:
+            print('{t} geopackage \nwritten to: {p}\ngenerated from: {q}'.format(t=self.name_extension(self.gpkg_name),
+                                                                                p=self.path,
+                                                                                q=self.table))
+        else:
+            print(u'{t} geopackage \nwritten to: {p}\ngenerated from: {q}'.format(
+                t=self.name_extension(self.gpkg_name),
+                p=self.path,
+                q=self.query))
+            
+    def shp_to_gpkg(self, shp_name=None, gpkg_name=None, tbl_name=None, print_cmd=False):
+        
+        """
+        Converts a Shapefile to a Geopackage file
+        :param gpkg_name: filename for geopackage (should end in .gpkg)
+        :param shp_name: filename for shape file (should end in .shp)
+        :param print_cmd: Print command line query
+        """
+        
+        assert gpkg_name[-5:] == '.gpkg', "The input file should end with .gpkg . Please check your input."
+        assert shp_name[-4:] == '.shp', "The input file should end with .shp . Please check your input."
+
+        # same function as shape file
+        self.path, shp_name = parse_shp_path(self.path, shp_name)
+
+        if not self.path:
+            self.path = file_loc('folder')
+
+        self.cmd = WRITE_SHP_CMD_GPKG.format(gpkg_name = gpkg_name,
```

Comment:
seems when writing table we don't use the param (tbl_name)? What is the usual value of self.table?